### ISSUE TITLE
Add 'Show Password' Button for Password Fields During Password Change Process

### ISSUE THREAD

**Issue by izabellaalves** [wagtail/wagtail#11948]
### Is your proposal related to a problem?

While changing my password, I noticed that it's not possible to see the last letter I typed, which is common in password fields, and there's also no "show password" button.

### Describe the solution you'd like

I would like a button next to the password-related fields where, upon clicking, the characters that were previously obscured can be seen, and upon clicking again, they become obscured.

![image](https://github.com/wagtail/wagtail/assets/97957709/d0048e98-714b-4fd7-98bb-4ff3ad6f5286)

Anyone can contribute to this. View our [contributing guidelines](https://docs.wagtail.org/en/latest/contributing/index.html), add a comment to the issue once you’re ready to start.

**Comment by gasman** [comment]
> I noticed that it's not possible to see the last letter I typed, which is common in password fields

Are you sure about this? I've only ever seen this behaviour on mobile devices, and indeed Wagtail's password fields do work that way on mobile:
![IMG_7044](https://github.com/wagtail/wagtail/assets/85097/3f4782ef-b23e-4d98-b36a-e6b907540847)

As far as I'm aware, this behaviour is always provided by the browser - it isn't something that Wagtail could implement on its own if the browser doesn't natively behave that way.

Happy to accept the feature request for a "show password" button, though.

**Comment by gasman** [comment]
Probably worth mentioning, there are known security issues with 'reveal password' features, since browsers treat them as plain text inputs while they're in the "visible" state, and may store or transmit that text insecurely for purposes like spellchecking and crash recovery:
https://www.bleepingcomputer.com/news/security/google-microsoft-can-get-your-passwords-via-web-browsers-spellcheck/
https://www.halborn.com/disclosures/demonic-vulnerability

It seems there's also some disagreement between security auditors and accessibility advocates about whether password reveal is good practice or not...
https://www.reddit.com/r/sysadmin/comments/xlgplm/auditors_say_password_reveal_button_is_a_security/

**Comment by zerolab** [comment]
It is interesting to see that GDS has now an official [password input component](https://design-system.service.gov.uk/components/password-input/) that does show/hide and it toggles beteewn `<input type="password"/>` and `<input type="text"/>`. 

Similar security concerns were raised in https://github.com/alphagov/govuk-design-system-backlog/issues/240 (which is the discussion for the new component)

**Comment by riaan-attar** [comment]
I had added the show password button the pull had been requested

**Comment by 29deepanshutyagi** [comment]
i want to work on this issue , if this issue is still opens @gasman , i will solve this by implementing a button next to password-related fields that toggles the visibility of the password when clicked. The password characters will switch between being obscured (<input type="password">) and visible (<input type="text">).

if any other approach is required to solve kindly gave me the details about that approach as well

**Comment by gasman** [comment]
Hi @29deepanshutyagi - thanks for your interest! There's an existing pull request for this issue at #11961, but it requires some extra work. If you're interested in picking this up, please see @lb-'s comment at https://github.com/wagtail/wagtail/pull/11961#issuecomment-2121160844.

**Comment by 29deepanshutyagi** [comment]
yes , i am willing to do that extra work and solve this issue completely

**Comment by lb-** [comment]
@29deepanshutyagi here's a good starting point for a Stimulus implementation of this feature. https://github.com/stimulus-components/stimulus-password-visibility/blob/master/src/index.ts

I'd recommend we add this capability to the the [ActionController](https://github.com/wagtail/wagtail/blob/5f158de94a8e16a7e7bea0795d0704c2ba5e2f94/client/src/controllers/ActionController.ts#L50). Maybe we should have a seperate controller, see what feels right when building it out. 

Also, a reminder that the issue is about the password field that's used when editing your account, not for the admin sign in form. 

I would not suggest this feature for the sign in form, based on the discussion above. However, once have the ability to use this for the password change form we can explore other usage.
